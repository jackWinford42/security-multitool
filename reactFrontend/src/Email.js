import React, { useState } from "react";
import { Alert, Card, CardBody} from 'reactstrap';
import RamtApi from "./Api";
//import axios from "axios";

/** Render the email page and handle a call to the emailrep api
 */
export default function Email() {
  console.debug("EMAIL COMPONENT")
  const [email, setEmail] = useState("");
  const [errors, setErrors] = useState([]);

  async function handleSubmit(evt) {
    evt.preventDefault();
    console.log(email)
    //const res = await axios(`https://emailrep.io/${email}?summary=true`);
    const res = await RamtApi.getEmailRes(email)
    console.log(res);
    // if (res.worked) history.push("/");
    // else setErrors(res.errors);
  }

  // Update form data to reflect change in form fields
  function handleChange(evt) {
    const { value } = evt.target;
    setEmail(value)
  }

	return (
		<div className="Email">
      <Card>
        <CardBody>
          <p>Enter an email to test its reputation.</p>
          <form className="form-inline" onSubmit={handleSubmit}>
            <label>Email:</label>
            <input
              className="form-control form-control-md"
              name="email"
              placeholder="email"
              value={email}
              onChange={handleChange}
            />
            <button type="submit" className="authButton btn btn-lg btn-primary">
              Investigate
            </button>
          </form>
          {errors.length ? <Alert color="danger">{errors}</Alert>:null}
        </CardBody>
      </Card>
		</div>
	);
}