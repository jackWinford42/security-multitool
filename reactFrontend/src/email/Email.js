import React, { useState } from "react";
import { Alert, Card, CardBody} from 'reactstrap';
import RamtApi from "../Api";
import Response from "./Response";
//import axios from "axios";

/** Render the email page and handle a call to the emailrep api
 */
export default function Email() {
  console.debug("EMAIL COMPONENT")
  const [email, setEmail] = useState("");
  const [data, setData] = useState({});

  async function handleSubmit(evt) {
    evt.preventDefault();
    const res = await RamtApi.getEmailRes(email)
    console.log("CHECKPOINT")
    console.log(res);
    setData(res)
  }

  // Update form data to reflect change in form fields
  function handleChange(evt) {
    const { value } = evt.target;
    setEmail(value)
  }

	return (
		<div className="Email">
      <Card>
        <CardBody>
          <p>Enter an email to test its reputation.</p>
          <form className="form-inline" onSubmit={handleSubmit}>
            <label>Email:</label>
            <input
              className="form-control form-control-md"
              name="email"
              placeholder="email"
              value={email}
              onChange={handleChange}
            />
            <button type="submit" className="authButton btn btn-lg btn-primary">
              Investigate
            </button>
          </form>
        </CardBody>
      </Card>
      <Response data={data}/>
		</div>
	);
}