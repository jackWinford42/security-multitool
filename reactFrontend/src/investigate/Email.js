import React, { useState } from "react";
import { useSelector } from "react-redux";
import { Card, CardBody } from 'reactstrap';
import RamtApi from "../common/Api";
import Response from "./Response";
import CircularProgress from '@mui/material/CircularProgress';
import "./Investigate.css"

/** Render the email page and handle a call to the emailrep api
 */
export default function Email() {
  console.debug("EMAIL COMPONENT")
  const userEmail = useSelector(st => st.currUser.email);
  const [email, setEmail] = useState("");
  const [data, setData] = useState({});
  const [loading, setLoading] = useState(false)

  async function handleSubmit(evt) {
    evt.preventDefault();
    setLoading(true);
    const res = (await RamtApi.investigate({type: "email", investigate: email, email: userEmail})).data
    setLoading(false);
    setData(res);
  }

  // Update form data to reflect change in form fields
  function handleChange(evt) {
    const { value } = evt.target;
    setEmail(value)
  }

	return (
		<div className="Email">
      <Card className="formCard">
        <CardBody>
          <p>Enter an email to test its reputation.</p>
          <small>Emails which are 100% safe are only saved to
            your history and are excluded from the site-wide history.</small>
          <form className="form-inline" onSubmit={handleSubmit}>
            <label>Email:</label>
            <input
              className="form-control form-control-md"
              name="email"
              placeholder="email"
              value={email}
              onChange={handleChange}
            />
            <button type="submit" className="authButton btn btn-lg btn-primary">
              Investigate
            </button>
          </form>
        </CardBody>
      </Card>
      {loading && <CircularProgress color="secondary" />}
      {data.message && <Response data={data}/>}
		</div>
	);
}