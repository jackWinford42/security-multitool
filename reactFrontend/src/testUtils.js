import React from "react";
import { Provider, useDispatch } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducers/rootReducer";
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from "redux-thunk";

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(thunk)));

const demoUser = {
  email: "jackwinford@gmail.com",
  username: "captainjack",
  password: "qwerty",
  photo_url: "https://bit.ly/3C2VUgQ",
};

// const dispatch = useDispatch();
// await dispatch({type: "BEGIN_AUTH_SESSION", user: demoUser});

const UserProvider =(
    ({children}) =>
    (<Provider store={store}>
      {children}
    </Provider>)
);

export { UserProvider };