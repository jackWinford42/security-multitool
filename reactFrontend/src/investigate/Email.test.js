import React from "react";
import { render } from "@testing-library/react";
import Email from "./Email";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils"

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Email/>
      </UserProvider>
    </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Email/>
      </UserProvider>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});