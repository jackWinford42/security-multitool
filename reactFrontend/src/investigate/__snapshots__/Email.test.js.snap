// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Email"
  >
    <div
      class="formCard card"
    >
      <div
        class="card-body"
      >
        <p>
          Enter an email to test its reputation.
        </p>
        <small>
          Emails which are 100% safe are only saved to your history and are excluded from the site-wide history.
        </small>
        <form
          class="form-inline"
        >
          <label>
            Email:
          </label>
          <input
            class="form-control form-control-md"
            name="email"
            placeholder="email"
            value=""
          />
          <button
            class="authButton btn btn-lg btn-primary"
            type="submit"
          >
            Investigate
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
